chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2240 @@
+
+        try:
+            await self.generate_new_dao(
+                amount_of_cats,
+                fee=fee,
+            )
+        except Exception as e_info:
+            await wallet_state_manager.user_store.delete_wallet(self.id())
+            self.log.error(f"Failed to create dao wallet: {e_info}")
+            raise
+
+        await self.wallet_state_manager.add_new_wallet(self)
+
+        # Now the dao wallet is created we can create the dao_cat wallet
+        cat_wallet: CATWallet = self.wallet_state_manager.wallets[self.dao_info.cat_wallet_id]
+        cat_tail = cat_wallet.cat_info.limitations_program_hash
+        new_dao_cat_wallet = await DAOCATWallet.get_or_create_wallet_for_cat(
+            self.wallet_state_manager, self.standard_wallet, cat_tail.hex()
+        )
+        dao_cat_wallet_id = new_dao_cat_wallet.wallet_info.id
+        dao_info = dataclasses.replace(self.dao_info, cat_wallet_id=cat_wallet.id(), dao_cat_wallet_id=dao_cat_wallet_id)
+        await self.save_info(dao_info)
+
+        return self
+
+    @staticmethod
+    async def create_new_dao_wallet_for_existing_dao(
+        wallet_state_manager: Any,
+        main_wallet: Wallet,
+        treasury_id: bytes32,
+        filter_amount: uint64 = uint64(1),
+        name: Optional[str] = None,
+    ) -> DAOWallet:
+        """
+        Create a DAO wallet for existing DAO
+        :param wallet_state_manager: Wallet state manager
+        :param main_wallet: Standard wallet
+        :param treasury_id: The singleton ID of the DAO treasury coin
+        :param filter_amount: Min votes to see proposal (user defined)
+        :param name: Wallet name
+        :return: DAO wallet
+        """
+        self = DAOWallet()
+        self.wallet_state_manager = wallet_state_manager
+        if name is None:
+            name = self.generate_wallet_name()
+
+        self.standard_wallet = main_wallet
+        self.log = logging.getLogger(name if name else __name__)
+        self.log.info("Creating DAO wallet for existent DAO ...")
+        self.dao_info = DAOInfo(
+            treasury_id=treasury_id,
+            cat_wallet_id=uint32(0),
+            dao_cat_wallet_id=uint32(0),
+            proposals_list=[],
+            parent_info=[],
+            current_treasury_coin=None,
+            current_treasury_innerpuz=None,
+            singleton_block_height=uint32(0),
+            filter_below_vote_amount=filter_amount,
+            assets=[],
+            current_height=uint64(0),
+        )
+        info_as_string = json.dumps(self.dao_info.to_json_dict())
+        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
+            name, WalletType.DAO.value, info_as_string
+        )
+        await self.resync_treasury_state()
+        await self.wallet_state_manager.add_new_wallet(self)
+        await self.save_info(self.dao_info)
+        if self.wallet_info is None:
+            raise ValueError("Internal Error")
```

Comment:
why would this happen?

Reply:
good point. The create_wallet method will raise a ValueError if wallet_info is none anyway, so this check is redundant.